backend/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM eclipse-temurin:17-jdk-alpine as build
+FROM eclipse-temurin:17.0.8_7-jdk as build
```

Comment:
whats the reason for this change?

Reply:
generally speaking alpine images tend to be leaner, and that speeds up our pipeline a bit
![image](https://github.com/CDCgov/prime-simplereport/assets/4952042/d92ca5a2-f552-4dea-afc7-d38f1ebc4984)

Reply:
The alpine image only supports linux/amd64 builds. This new one supports several other architectures.

The new most recent [backend build on this PR took `2m 18s`](https://github.com/CDCgov/prime-simplereport/actions/runs/6187219669/usage?pr=6457) while the most recent [backend build on a recent queued PR took `2m 57s`](https://github.com/CDCgov/prime-simplereport/actions/runs/6225658032/usage) so I don't think build time has been an issue I've noticed. 

A lightly large image is something we'll need to keep an eye on and I generally agree that keeping the images small is a priority, that said, the [E2E tests on this PR took `7m 30s`](https://github.com/CDCgov/prime-simplereport/actions/runs/6187219669/usage?pr=6457) and [`7m 51s` on the queued branch](https://github.com/CDCgov/prime-simplereport/actions/runs/6225658032/usage) I mentioned.